include/revng/Pipeline/Structure.h

```diff
@@ -0,0 +1,224 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <initializer_list>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/iterator_range.h"
+#include "llvm/Support/Casting.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "revng/Pipeline/Errors.h"
+#include "revng/Pipeline/Step.h"
+
+namespace Pipeline {
+
+class PipelineExecutionEntry {
+public:
+  Step *ToExecute;
+  ContainersStatus Objectives;
+
+  PipelineExecutionEntry(Step &ToExecute, ContainersStatus Objectives) :
+    ToExecute(&ToExecute), Objectives(std::move(Objectives)) {}
+};
+
+class BoundedStep {
```

Comment:
1. The past participle of *bind* is *bound*.
2. Can we squash this class in `Step`? It's an extra layer layer bringing very little benefit.
3. Can you justify the usage of a `std::string` as opposed to a `Step *`?

Reply:
3 was because i was fearful of iterator invalidation, which should not happen for string map, i will use Step* but it is spooky.